suplemon/hex2xterm.py

```diff
@@ -0,0 +1,23 @@
+
+# Default color levels for the color cube
+cubelevels = [0x00, 0x5f, 0x87, 0xaf, 0xd7, 0xff]
+# Generate a list of midpoints of the above list
+snaps = [(x+y)/2 for x, y in list(zip(cubelevels, [0]+cubelevels))[1:]]
+
+
+def hex_to_rgb(value):
+    value = value.lstrip("#")
+    lv = len(value)
+    return tuple(int(value[i:i + lv // 3], 16) for i in range(0, lv, lv // 3))
+
+
+def hex_to_xterm(hex):
+    """
+    Converts hex color values to the nearest equivalent xterm-256 color.
+    """
+    rgb = hex_to_rgb(hex)
+    r, g, b = rgb
+    # Using list of snap points, convert RGB value to cube indexes
+    r, g, b = map(lambda x: len(tuple(s for s in snaps if s < x)), (r, g, b))
+    # Simple colorcube transform
+    return r*36 + g*6 + b + 16
```

Comment:
Ah, I thought these were at render time. Load time of the theme should work as well :+1: